src/formats/roam-json.ts

```diff
@@ -255,8 +307,45 @@ export class RoamJSONImporter extends FormatImporter {
 		return blockText;
 	};
 
-	private async jsonToMarkdown(graphFolder: string, attachmentsFolder: string, json: RoamPage | RoamBlock, indent: string = '', isChild: boolean = false): Promise<string> {
+	// setup to hold the newest and oldest timestamp value from a given page
+	newestTimestamp: number = 0;
+	oldestTimestamp: number = 0;
+
+	private async jsonToMarkdown(graphFolder: string, attachmentsFolder: string, json: RoamPage | RoamBlock, indent: string = '', isChild: boolean = false, setTitleProperty: string, createdTimestamp: number, updatedTimestamp: number): Promise<string> {
 		let markdown: string[] = [];
+		let frontMatterYAML: string[] = [];
+		// use Roam's create-time and edit-time values to set timestamps
+		const jsonEditTime = json['edit-time'];
+		const jsonCreateTime = json['create-time'];
+		let checkYAMLoptions: boolean = false;
+
+		// for YAML frontmatter
+		// can't be edited before it was created, compare timestamps
+		if (this.newestTimestamp < this.oldestTimestamp) {
+			this.oldestTimestamp = this.newestTimestamp;
+		}
+
+		// check the edit-time of the block, compare to what was passed, use the most recent date
+		// if undefined, set newestTimestamp to the value of updatedTimestamp
+		this.newestTimestamp = (!jsonEditTime || updatedTimestamp > jsonEditTime)
+    		? updatedTimestamp
+    		: jsonEditTime;
+
+		// if the create time is defined, set oldestTimestamp to the lower of the createdTimestamp value or jsonCreateTime
+		// else, set oldestTimestamp to the value of createdTimestamp
+		if (jsonCreateTime !== undefined) {
+			if (createdTimestamp > 10) { // passed as a 0
+				this.oldestTimestamp = (createdTimestamp < jsonCreateTime) 
```

Comment:
Oh actually I think we could make this even simpler.

```
this.oldestTimestamp = Math.min(createdTimestamp, jsonCreateTime);
```

Reply:
That's much cleaner, much appreciated!